### PR TITLE
Add yang model definition for VOQ_INBAND_INTERFACE

### PR DESCRIPTION
**Pull request by tjchadaga**
#### Why I did it
Add yang model definition for VOQ_INBAND_INTERFACE defined and implemented for VOQ chassis. HLD for voq-inband-interface is included in https://github.com/sonic-net/SONiC/blob/master/doc/voq/voq_hld.md

#### How I did it
Added yang model definition, unit tests, sample config and documentation for the table

#### How to verify it
1. Validated config tree generation using "pyang -Vf tree -p /usr/local/share/yang/modules/ietf ./yang-models/sonic-voq-inband-interface.yang"

2. Built the below python-wheels to validate unit tests and other changes
target/python-wheels/bullseye/sonic_yang_mgmt-1.0-py3-none-any.whl 
target/python-wheels/bullseye/sonic_yang_models-1.0-py3-none-any.whl 
target/python-wheels/bullseye/sonic_config_engine-1.0-py3-none-any.whl

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [x] 202205

#### Link to config_db schema for YANG module changes

https://github.com/Azure/sonic-buildimage/blob/6fd7eab250ac54e9fedb68fc5248fdef2ba0adf6/src/sonic-yang-models/doc/Configuration.md#voq-inband-interface

### CODE REVIEW

**Review by arlakshm — APPROVED** [review]
lgtm

File: src/sonic-yang-models/yang-models/sonic-voq-inband-interface.yang [hunk 1]
```diff
@@ -0,0 +1,56 @@
+module sonic-voq-inband-interface {
+    namespace "http://github.com/Azure/sonic-voq-inband-interface";
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-voq-inband-interface.yang:2]: Use sonic-net to replace Azure?

File: src/sonic-yang-models/yang-models/sonic-voq-inband-interface.yang [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                key "name";
+                leaf name {
+                    type string {
+                        pattern "Ethernet-IB[0-9]+";
+                    }
+                }
+                leaf inband_type {
+                    description "Type of inband interface";
+                    type string {
+                        pattern "port|Port";
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-voq-inband-interface.yang:37]: what's the difference between port and Port?

Comment (tjchadaga) [reply src/sonic-yang-models/yang-models/sonic-voq-inband-interface.yang:37]: Just the case

File: src/sonic-yang-models/yang-models/sonic-voq-inband-interface.yang (hunk 1 above)
Comment (tjchadaga) [reply src/sonic-yang-models/yang-models/sonic-voq-inband-interface.yang]: done

**Review by ganglyu — APPROVED** [review]

**Comment by zhangyanzhao** [comment]
@xumia can you please help to check the build failure? Thanks.

**Comment by yxieca** [comment]
@tjchadaga can you raise separate PR for 202205 branch or suggest dependency?

**Comment by tjchadaga** [comment]
> @tjchadaga can you raise separate PR for 202205 branch or suggest dependency?

@yxieca - Raised https://github.com/sonic-net/sonic-buildimage/pull/12493 for 202205

**Pull request merged**